File: data_structures/arrays/0001ReverseAnArray/reverse.py
def reverse_list(array: list, start_index: int, end_index: int) -> None:
    while start_index < end_index:
        array[start_index], array[end_index] = array[end_index], array[start_index]
        start_index += 1
        end_index -= 1  


def main():
    array = [1, 2, 3, 4, 5]
    start_index = 0
    end_index = len(array) - 1
    print("The Original Array Is: ", array)
    reverse_list(array, start_index, end_index)
    print("Reversed Array Is: ", array)

if __name__ == "__main__":
    main()

File: data_structures/arrays/0001ReverseAnArray/reverse.cpp
#include <bits/stdc++.h>

using namespace std;

void swap_numbers(int *first, int *second) {
  int temp = *first;
  *first = *second;
  *second = temp;
}

void print_array(int array[], int size) {
  for (int i = 0; i < size; i++) cout << array[i] << " ";

  cout << endl;
}

void reverse_array(int array[], int start_index, int end_index) {
  while (start_index < end_index) {
    swap_numbers(&array[start_index], &array[end_index]);
    start_index++;
    end_index--;
  }
}

int main() {
  int array[] = {1, 2, 3, 4, 5};
  int start_index = 0;
  int end_index = sizeof(array) / sizeof(array[0]) - 1;
  reverse_array(array, start_index, end_index);
  print_array(array, end_index + 1);
}